Document workaround for using HashRouter (#69140)
elastic/kibana: src/core/MIGRATION.md

```

### Logging config migration

[Read](./server/logging/README.md#logging-config-migration)

### Use HashRouter in migrated apps

Kibana applications are meant to be leveraging the `ScopedHistory` provided in an app's `mount` function to wire their router. For react,

> this is done by using the `react-dom` `Router` component:

Reviewer: ```suggestion
this is done by using the `react-router-dom` `Router` component:
```

```typescript
export const renderApp = async (element: HTMLElement, history: ScopedHistory) => {
  render(
    <I18nProvider>
      <Router history={history}>
        <Switch>
          <Route path="/some-route/:id" render={renderTutorial} />
          <Route path="/some-other-route/:tab?" render={renderTutorialDirectory} />
        </Switch>
      </Router>
    </I18nProvider>,
    element
  );

  return () => {
    unmountComponentAtNode(element);
    unlisten();
  };
};
```

> Some legacy apps were using react's `HashRouter` instead. Using `HashRouter` in a migrated application will cause some route change 

Reviewer: nit
```suggestion
Some legacy apps were using react-router-dom's `HashRouter` instead. Using `HashRouter` in a migrated application will cause some route change 
```

events to not be catched by the router, as the `BrowserHistory` used behind the provided scoped history does not emit 
the `hashevent` that is required for the `HashRouter` to behave correctly.

> It is strictly recommended to migrate your application's routing to browser history, which is the only routing officially supported by the platform.

Reviewer: Maybe we add a strong warning here:
```suggestion
It is strictly recommended to migrate your application's routing to browser history, which is the only routing officially supported by the platform. You may use this workaround at your own risk. While we are not aware of any problems it currently creates, there may be edge cases that do not work properly.
```
